### PR TITLE
Add bearer authentication

### PR DESCRIPTION
**Pull request by danielgattiluxor**
Add bearer authentication

### CODE REVIEW

File: trino/trino.go [hunk 1]
```diff
@@ -130,6 +130,8 @@ const (
 	trinoAddedPrepareHeader       = trinoHeaderPrefix + `Added-Prepare`
 	trinoDeallocatedPrepareHeader = trinoHeaderPrefix + `Deallocated-Prepare`
 
+	authorization = `Authorization`
```
Comment (nineinchnick) [inline trino/trino.go:133]: Isn't this supposed to be `Authentication`?

File: README.md [hunk 2]
```diff
@@ -71,6 +71,10 @@ This driver supports Kerberos authentication by setting up the Kerberos fields i
 
 Please refer to the [Coordinator Kerberos Authentication](https://trino.io/docs/current/security/server.html) for server-side configuration.
 
+#### Bearer authentication
+
+This driver supports Bearer authentication (also called token authentication) by setting it in the [Config](https://godoc.org/github.com/trinodb/trino-go-client/trino#Config) struct.
```
Comment (nineinchnick) [inline README.md:76]: This needs an example, especially if users are supposed to prefix any tokens with `Bearer`.

File: trino/trino.go [hunk 3]
```diff
@@ -227,6 +230,7 @@ func (c *Config) FormatDSN() (string, error) {
 		"session_properties": strings.Join(sessionkv, ","),
 		"extra_credentials":  strings.Join(credkv, ","),
 		"custom_client":      c.CustomClientName,
+		"bearer_auth":       c.BearerAuth,
```
Comment (nineinchnick) [inline trino/trino.go:233]: This needs a test, ideally an integration one, where you would configure the Trino server to accept this kind of authentication.

**Review by nineinchnick — CHANGES_REQUESTED** [review]

File: trino/trino.go (hunk 1 above)
Comment (reedip) [reply trino/trino.go:133]: No, this would be Authorization to match "Authorization: Bearer {token}"

File: trino/trino.go (hunk 3 above)
Comment (reedip) [reply trino/trino.go:233]: I would like this feature , if its not too much to ask, can I continue on this PR?

Comment (nineinchnick) [reply trino/trino.go:233]: Sure, you can copy this branch into your fork and open up a second PR.

File: trino/trino.go [hunk 4]
```diff
@@ -332,6 +336,7 @@ func newConn(dsn string) (*Conn, error) {
 		trinoSchemaHeader:          query.Get("schema"),
 		trinoSessionHeader:         query.Get("session_properties"),
 		trinoExtraCredentialHeader: query.Get("extra_credentials"),
+		authorization:              query.Get("bearer_auth"),
```
Comment (reedip) [inline trino/trino.go:339]: We need to change the code a bit in the line#342 as 

```
 if v != "" {
                    if k == "Authorization" {
                        c.httpHeaders.Add(k, fmt.Sprintf("Bearer %s", v))
                    }else {
                        c.httpHeaders.Add(k, v)
                    }
```

**Comment by nineinchnick** [comment]
Superseded by #113

**Pull request closed without merging**